src/PoolManager.sol

```diff
@@ -205,18 +252,45 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        if (key.hooks.shouldCallBeforeModifyPosition()) {
-            bytes4 selector = key.hooks.beforeModifyPosition(msg.sender, key, params, hookData);
+        if (key.hooks.shouldCallBeforeRemoveLiquidity()) {
+            bytes4 selector = key.hooks.beforeRemoveLiquidity(msg.sender, key, params, hookData);
             // Sentinel return value used to signify that a NoOp occurred.
             if (key.hooks.isValidNoOpCall(selector)) {
                 // We only want to clear the current hook if it was set in setCurrentHook in this execution frame.
                 if (set) Lockers.clearCurrentHook();
                 return BalanceDeltaLibrary.MAXIMUM_DELTA;
-            } else if (selector != IHooks.beforeModifyPosition.selector) {
+            } else if (selector != IHooks.beforeRemoveLiquidity.selector) {
+                revert Hooks.InvalidHookResponse();
+            }
+        }
+
+        delta = _modifyPosition(
+            key,
+            IPoolManager.PoolModifyPositionParams({
+                tickLower: params.tickLower,
+                tickUpper: params.tickUpper,
+                liquidityDelta: -params.liquidityDelta.toInt128()
```

Comment:
IMO this is no longer `liquidityDelta` since you're always accepting positive value and flipping it negative in the remove case, so is more like `liquidityToRemove`

Reply:
also -- you don't sanitize `liquidityDelta` to ensure positive here? so I can pass in negative value to `removeLiquidity` and it'll get negated so we'll actually end up removing liquidity :D

Reply:
and vice versa for addLiquidity, if you pass negative value it will actually remove hahah

Reply:
oh nvm.. you made liquidityDelta an unsigned int

Reply:
Pulled in your changes, so no longer relevant